syntax = "proto3";

// /usr/local/bin/protoc -I=. --cpp_out=. game.proto
// /usr/local/bin/protoc -I=. --grpc_out=. --plugin=protoc-gen-grpc=/home/dkwapisz/grpc/cmake/build/grpc_cpp_plugin game.proto
// python -m grpc_tools.protoc -I. --python_out=. --grpc_python_out=. ./game.proto

package GameMessage;

service StateActionExchange {
  rpc StateAction (State) returns (Action) {}

  rpc StateReset (State) returns (Reset) {}
}

message State {
  string mapMatrix = 1;
  int32 coinsLeft = 2;
  int32 reward = 3;
  int32 stepsCount = 4;
  bool gameOver = 5;
  int32 episodeCount = 6;
}

message Action {
  enum ActionDirection {
    IDLE = 0;
    UP = 1;
    RIGHT = 2;
    DOWN = 3;
    LEFT = 4;
  }

  ActionDirection moveDirection = 1;
  ActionDirection shotDirection = 2;
}

message Reset {
  bool resetNeeded = 1;
}